### DISCUSSION TITLE
Create faker instances while avoiding bug regarding too big bundle size

### DISCUSSION THREAD

**Discussion by tobiaswaltl** [faker-js/faker#2178] [Q&A] [answered]
Hi,

since faker v8 the property `locales` is private.
We have the need to create new faker instances with specific locales while also avoiding the [bug](https://github.com/faker-js/faker/issues/1791) regarding the large bundle size.
Previously, we achieved that via deep importing the locale-specific faker (to avoid the bundle size bug) and then creating a new faker instance using the locales of the exported one. As `locales` is private now, this approach doesn't work anymore.
Is there a way to e.g. deep-import the locale definitions directly and passing them to the `locale` field in the constructor?

Here's our previous approach for v7:
```
import { faker as defaultDeFaker } from '@faker-js/faker/locale/de';

const newFaker = new Faker({
  locales: defaultDeFaker.locales,
});
```

**Comment by Shinigami92** [comment, marked as answer]
So I assume this is not working for you, due to the bug? https://fakerjs.dev/guide/localization.html#custom-locales-and-fallbacks

Could you try this and report us back?

```ts
import { Faker } from "@faker-js/faker";
import { faker as fakerDe } from "@faker-js/faker/locale/de";

const myCustomFaker = new Faker({
  locale: [fakerDe.rawDefinitions, ...],
});
```

**Reply by tobiaswaltl** [reply]
Right. Thank you for your suggestion. It is exactly what I was looking for!

**Comment by ST-DDT** [comment]
> Here's our previous approach for v7:
> 
> ```
> import { faker as defaultDeFaker } from '@faker-js/faker/locale/de';
> 
> const newFaker = new Faker({
>   locales: defaultDeFaker.locales,
> });
> ```

I would like to understand what do you wish to achieve with this code?
Isnt the result the same as `defaultDeFaker`?
